pkg/controller/studyjob/studyjob_controller.go

```diff
@@ -393,46 +394,55 @@ func (r *ReconcileStudyJobController) updateWorker(c katibapi.ManagerClient, ins
 	return update, nil
 }
 
-func (r *ReconcileStudyJobController) getJobWorkerStatus(w *katibv1alpha1.WorkerCondition, ns string) WorkerStatus {
-	runtimejob := createWorkerJobObj(w.Kind)
-	nname := types.NamespacedName{Namespace: ns, Name: w.WorkerID}
-	joberr := r.Client.Get(context.TODO(), nname, runtimejob)
-	if joberr != nil {
-		return WorkerStatus{}
-	}
+func (r *ReconcileStudyJobController) getJobWorkerStatus(ns string, wid string, wkind *schema.GroupVersionKind) WorkerStatus {
+	nname := types.NamespacedName{Namespace: ns, Name: wid}
 	var state katibapi.State = katibapi.State_RUNNING
 	var cpTime *metav1.Time
-	switch w.Kind {
+	switch wkind.Kind {
+
 	case DefaultJobWorker:
```

Comment:
Yes. But the string values are different currently.  'Complete' in  https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/batch/v1/types.go#L170  vs 'Succeeded' in https://github.com/kubeflow/tf-operator/blob/master/pkg/apis/common/v1beta2/common_types.go#L121 

In general, since 'job' is a built-in K8s  Kind, should we care about it matching with our JobCondition?  I feel that the only necessary condition is that  all custom job operators share the same JobStatus type. 

Or other option is, we can change the common JobConditionType string to match the batch Job's JobConditionType. But new developers might not be aware of this assumption and may break the logic.

Reply:
I think this is ok for this version. One thing that we have discussed in the past is moving the common types into its own repo. If we go down that route, it will be easier to change then. What do you think?

Reply:
Yup. There is a discussion in https://github.com/kubeflow/mpi-operator/issues/92#issuecomment-464777479